modules/ocl/cv_calibration_parser.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * cv_calibration_parser.h - parse fisheye calibration file
+ *
+ *  Copyright (c) 2016-2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Junkai Wu <junkai.wu@intel.com>
+ */
+
+#ifndef XCAM_CV_CALIBRATION_PARSER_H
+#define XCAM_CV_CALIBRATION_PARSER_H
+
+#include "xcam_utils.h"
+#include <vector>
+
+namespace XCam {
+
+struct IntrinsicParameter {
+    float xc;
+    float yc;
+    float c;
+    float d;
+    float e;
+    int poly_length;
+
+    std::vector<float> poly_coeff;
+};
+
+struct ExtrinsicParameter {
+    float trans_x;
+    float trans_y;
+    float trans_z;
+    float roll;
+    float pitch;
+    float yaw;
+};
+
+class CalibrationParser
+{
+
+public:
+    explicit CalibrationParser ();
+
+    IntrinsicParameter getIntrinsicParam(char *file_str);
+    ExtrinsicParameter getExtrinsicParam(char *file_str);
+
+private:
+    XCAM_DEAD_COPY (CalibrationParser);
+ 
+    void parseIntrinsicParam(char *file_str);
+    void parseExtrinsicParam(char *file_str);
+

```

Comment:
code style:
1. function names and variable names. (_intrinsicParam=> _intrinsic_param)
2. parse_xxx need return values. anything parsed error, it's better to add a log XCAM_LOG_WARNING/ERROR or use
XCAM_FAIL_RETURN to check status and output log.

e.g.


     XCamReturn parse_intrinsic_param(char *body,  IntrinsicParameter &param);
    ...
     //IntrinsicParameter _intrinsic_param;